setup.sh

```diff
@@ -9,6 +9,12 @@ FLUTTER_VERSION=${1:-3.0.2}
 FLUTTER_CHANNEL=${2:-stable}
 FLUTTER_OS=$OS
 
+# Detect the latest version
+if [[ $FLUTTER_VERSION == latest ]]
+then
+        FLUTTER_VERSION=$(curl -s https://storage.googleapis.com/flutter_infra_release/releases/releases_$OS.json | jq -r ".releases | map(select(.channel == \"${FLUTTER_CHANNEL}\")) | .[0].version")
```

Comment:
I think we can narrow down the release via the `current_release` object as it contain the `hash` for the release of all available channels.

Reply:
Maybe we can do sth like this

```
# Detect the latest version
if [[ $FLUTTER_VERSION == "latest" ]]
then
        echo "Detecting latest version..."
        curl -L https://storage.googleapis.com/flutter_infra_release/releases/releases_$OS.json -o /tmp/flutter_release.json
        CURRENT_RELEASE=$(jq -r ".current_release.${FLUTTER_CHANNEL}" /tmp/flutter_release.json)
        FLUTTER_VERSION=$(jq -r ".releases | map(select(.hash == \"${CURRENT_RELEASE}\")) | .[0].version" /tmp/flutter_release.json)
        rm /tmp/flutter_release.json
fi
```